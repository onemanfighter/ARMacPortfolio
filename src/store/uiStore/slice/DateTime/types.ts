export type DateTimeState = {
  date?: Date;
};

export interface DateTimeActions {
  /**
   * Triggers the timer to update the time & date every second.
   */
  init: () => void;
}

export type DateTimeSlice = DateTimeState & DateTimeActions;
