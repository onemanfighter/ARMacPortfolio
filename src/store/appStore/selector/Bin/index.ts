export { binSelector } from './Bin.selector';
