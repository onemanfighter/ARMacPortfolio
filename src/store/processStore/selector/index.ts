export { useShallow } from 'zustand/react/shallow';

export * from './Power';
export * from './Login';
export * from './ActiveApp';
