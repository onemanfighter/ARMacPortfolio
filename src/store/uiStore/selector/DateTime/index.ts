export { dateTimeSelector } from './DateTime.selector';
