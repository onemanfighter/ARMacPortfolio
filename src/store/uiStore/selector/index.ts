export { useShallow } from 'zustand/react/shallow';

export * from './Modal';
export * from './DateTime';
