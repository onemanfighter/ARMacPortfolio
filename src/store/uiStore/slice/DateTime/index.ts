export { default as createDateTimeSlice } from './DateTime.slice';
export type { DateTimeSlice, DateTimeState } from './types';
