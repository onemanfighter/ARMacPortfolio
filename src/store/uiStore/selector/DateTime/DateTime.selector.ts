import { UiStoreState } from '../../uiStore';
import _ from 'lodash';

const dateTimeSelector = (state: UiStoreState) => ({
  date: state.DateTime.date?.toDateString(),
  dateWithoutYear: state.DateTime.date?.toDateString().slice(0, -5),
  time: state.DateTime.date?.toLocaleTimeString(),
  timeInAmPm: () => {
    const time = state.DateTime.date
      ?.toLocaleTimeString()
      .split(':')
      .slice(0, 2)
      .join(':');
    const hour = _.toNumber(time?.split(':')[0]);
    if (hour > 12) return `${hour - 12}:${time?.split(':')[1]} PM`;

    return `${time} AM`;
  },
  init: state.DateTime.init,
});

export { dateTimeSelector };
