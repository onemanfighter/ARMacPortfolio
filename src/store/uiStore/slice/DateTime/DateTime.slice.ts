import { UiStoreSlice } from '../../uiStore';
import { DateTimeSlice, DateTimeState } from './types';

const defaultDateTimeState: DateTimeState = {
  date: new Date(),
};

const createDateTimeSlice: UiStoreSlice<DateTimeSlice> = (set) => ({
  ...defaultDateTimeState,
  init: () => {
    setInterval(() => {
      set((state) => {
        state.DateTime.date = new Date();
      });
    }, 1000);
  },
});

export default createDateTimeSlice;
