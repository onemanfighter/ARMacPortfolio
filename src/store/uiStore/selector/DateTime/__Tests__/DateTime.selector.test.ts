import { renderHook } from '@testing-library/react';
import { uiStore } from '@uiStore';
import { dateTimeSelector } from '../DateTime.selector';

describe('Date time selector', () => {
  it('should return date time selector state and actions', () => {
    const dateTime = renderHook(() => uiStore(dateTimeSelector)).result.current;

    expect(dateTime).toMatchSnapshot();
  });
});
