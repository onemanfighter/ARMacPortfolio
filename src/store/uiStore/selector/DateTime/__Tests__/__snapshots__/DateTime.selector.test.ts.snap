// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Date time selector should return date time selector state and actions 1`] = `
{
  "date": "Thu May 30 2024",
  "dateWithoutYear": "Thu May 30",
  "init": [Function],
  "time": "2:21:57 AM",
  "timeInAmPm": [Function],
}
`;
